meteor/client/lib/ModalDialog.tsx

```diff
@@ -307,15 +307,15 @@ class ModalDialogGlobalContainer0 extends React.Component<
 	}
 	onAccept = (e: SomeEvent, inputResult: ModalInputResult) => {
 		const queue = this.state.queue
-		const onQueue = queue.pop()
+		const onQueue = queue.shift()
```

Comment:
I take that taking the first instead of the last has no effect on the result?
`.pop()` takes the last element in the array, where `.shift()` takes the first.

Or have you also inverted the order of elements in the array prior to this event firing?

Reply:
This change was made because the queue was in fact working as a stack, which didn't make sense for pop-ups. Before, when experiencing multiple pop-ups, you would constantly see the first pop-up, and then remove the last one in the queue (one you would never get to actually interact with). Instead you would constantly interact with the first pop-up in the queue until this was the last one left.

Reply:
Ah so before it was "Last In, First out", and now it is "First in, First out". did i understand it right?

Reply:
Exactly, but that wasn't even visible to the user because the scenario never arose (as far as i know) before handling the multiple possible duplicates that can come on import now. So the user only ever saw the first modal dialog before the change, and every interaction just discarded the most recent addition without the user ever seeing it.